cranelift/fuzzgen/src/function_generator.rs

```diff
@@ -1224,16 +1259,19 @@ where
     ///
     /// `min_size`: Controls the amount of space that the address should have.This is not
     /// guaranteed to be respected
+    ///
+    /// `max_offset`: Limits the offset from the stack slot.
     fn generate_load_store_address(
         &mut self,
         builder: &mut FunctionBuilder,
         min_size: u32,
+        max_offset: u32,
```

Comment:
It confused me that this parameter is called `max_offset`, because the function returns a value named `offset`, but this parameter is not a maximum for that value.

How about an `aligned: bool` parameter instead? The caller can decide whether to set the `aligned` memflag, and inform this function of that choice; that would be good to test anyway. On aarch64 we'd force it to set that flag on atomic ops until #5483 gets fixed.

The effect of this parameter would be something like this:
```rust
            // stack_slot_with_size guarantees that slot_size >= min_size
            let max_offset = slot_size - min_size;
            let offset = if aligned {
                self.u.int_in_range(0..=max_offset / min_size)? * min_size
            } else {
                self.u.int_in_range(0..=max_offset)
            };
```
I'm assuming that stack slots are always sufficiently aligned, although I don't actually know if that's true.

Reply:
Yeah, I wasn't a great fan of the `max_offset` thing, but wasn't able to think of anything else at the time. But with the aligned flag and returning the max offset it turned out way better! Thanks! It's also really nice that we can neatly fit in some MemFlags testing as well.

I've had to change this so that we always align on 8 byte boundaries instead of `min_size` since it would cause unaligned addresses with `i8` and `i16`.

>I'm assuming that stack slots are always sufficiently aligned, although I don't actually know if that's true.

At least for AArch64 we are [aligning them at 16 byte boundaries](https://github.com/bytecodealliance/wasmtime/blob/44913825b5e93d40ea4ca4fb3d03ab3aa15e1714/cranelift/codegen/src/isa/aarch64/abi.rs#L111-L115), but I'm not sure what the requirements are for other arches.

Reply:
> I've had to change this so that we always align on 8 byte boundaries instead of `min_size` since it would cause unaligned addresses with `i8` and `i16`.

Oh, why do smaller values need more than their natural alignment? I'd expect `i16` to work at any even address, for instance. And in the other direction, don't `i128` values need 16-byte alignment?

Reply:
For some reason I had the idea that we would need a minimum of 8 byte alignment. But I've been reading the alignment requirements (for AArch64 atomics) and they do mention that it should be a multiple of the access size.